### PR TITLE
Worker registration (context) improvement

### PR DESCRIPTION
**Pull request by jrmbrgs**
* adding worker & job registration
 * adding log

### CODE REVIEW

File: Library/Context/ContextStorageService.php [hunk 1]
```diff
@@ -223,4 +223,61 @@ public function updateWorkflowStep(string $workflowProcessId, string $stepCode,
...
+                        ctx._source.workflow.steps[stepIndex]['jobList'][jobIndex] << jobHash
+                        break;
+                    }
+                }
+            }
+        }
+eot;
+        $updateHash = [
+            'script' => [
+                'lang' => 'groovy',
```
Comment (adefretin) [inline Library/Context/ContextStorageService.php:275]: could be set as const

File: Library/Core/AbstractWorker.php [hunk 2]
```diff
@@ -53,6 +53,8 @@
 
     protected $workflowConfig;
 
+    protected $workerHostname = '';
```
Comment (adefretin) [inline Library/Core/AbstractWorker.php]: missing comment

File: Library/Core/AbstractWorker.php [hunk 3]
```diff
@@ -70,6 +72,7 @@ protected function initWorker()
             $this->workflowConfig['servicesClassNameSpace'],
             $this->workflowConfig['servicesClassPath']
         );
+        $this->workerHostname = php_uname("n");
```
Comment (adefretin) [inline Library/Core/AbstractWorker.php]: use simple quote

File: Library/Core/AbstractWorker.php [hunk 4]
```diff
@@ -143,9 +146,10 @@ private function parseOpt(array $paramList)
                 }
                 switch ($msg->err) {
                     case '-191': // no more msg
+                    case '-185': // timeout
```
Comment (adefretin) [inline Library/Core/AbstractWorker.php:162]: set as const

File: Library/Core/Storage/ElasticsearchAdapter.php [hunk 5]
```diff
@@ -402,6 +404,9 @@ public function update(string $id, array $updateHash) : array
                 ['id' => $id],
                 $this->commonRequestParamHash
             );
+            if ($retryNb) {
+                $requestParamHash['retry_on_conflict'] = $retry;
```
Comment (adefretin) [inline Library/Core/Storage/ElasticsearchAdapter.php]: =>   = $retryNb

File: Library/Workflow/ManagerService.php [hunk 6]
```diff
@@ -136,6 +136,7 @@ public function getContext(string $workflowProcessId)
     public function setStatus(string $workflowProcessId, string $status, string $message = '')
     {
         $this->di->get('logr')->debug(json_encode(func_get_args()));
+        $this->di->get('logr')->info("Workflow $workflowProcessId is now $status");
```
Comment (adefretin) [inline Library/Workflow/ManagerService.php:163]: simple quote

File: Library/Workflow/ManagerService.php [hunk 7]
```diff
@@ -301,6 +307,7 @@ public function processStepJobResult(string $workflowProcessId, string $stepCode
         }
         $stepHash['jobList'][$jobId]['status'] = $resultHash['status'] ?? self::STATUS_FAILED;
         $stepHash['jobList'][$jobId]['result'] = $resultHash['data'] ?? [];
+        $stepHash['jobList'][$jobId]['finishedAt'] = date('Y-m-d H:i:s');
```
Comment (adefretin) [inline Library/Workflow/ManagerService.php]: set date format as const

File: Library/Workflow/ManagerWorker.php [hunk 8]
```diff
@@ -62,11 +62,14 @@ protected function initWorker()
      */
     protected function processMessage(Message\MessageDto $messageDto)
     {
-        $this->getDI()->get('logr')->info('messageDto : ' . $messageDto);
+        $this->getDI()->get('logr')->debug($messageDto);
         switch ($messageDto->getType()) {
             case Message\MessageDto::TYPE_WF_CTRL:
                 switch ($messageDto->getAction()) {
                     case 'start':
```
Comment (adefretin) [inline Library/Workflow/ManagerWorker.php]: add const

Comment (jrmbrgs) [reply Library/Workflow/ManagerWorker.php]: done

File: Library/Workflow/ManagerService.php (hunk 6 above)
Comment (jrmbrgs) [reply Library/Workflow/ManagerService.php:163]: needed

File: Library/Core/Storage/ElasticsearchAdapter.php (hunk 5 above)
Comment (jrmbrgs) [reply Library/Core/Storage/ElasticsearchAdapter.php]: THX ! done

File: Library/Core/AbstractWorker.php (hunk 4 above)
Comment (jrmbrgs) [reply Library/Core/AbstractWorker.php:162]: I'll check if rdkafka provides some

File: Library/Core/AbstractWorker.php (hunk 3 above)
Comment (jrmbrgs) [reply Library/Core/AbstractWorker.php]: done

**Review by adefretin — APPROVED** [review]

**Pull request merged**